Hakizimana-Clement/Advizr
# [Merge]: Develop Branch Updates

### PR Description
This PR merges the latest changes from the develop branch, incorporating the [Feature]: Update Username functionality, which allows users to edit and update their profile username. 

These updates have been tested and reviewed to ensure quality before merging into the production branch.

### What Does This PR Do?
This PR includes:
- The new Update Username feature that enables users to edit and save their usernames in the database.
- Improved user experience with real-time feedback during updates and robust error handling.
- Updates to the auth service to include a function for updating user profiles.
### Description of Key Updates
1. Feature Implementation
- A form is now available for users to view and update their usernames.
- Database updates for username changes are handled using React Query for seamless state management.
2. Error Handling
- Comprehensive error handling ensures that users are notified of issues such as invalid input or server errors.
3. Form Validation
- Username inputs are validated to meet predefined criteria (e.g., non-empty).
### How Should This Be Manually Tested?
1. Switch to the develop branch.
2. Test the username update feature by navigating to the user settings page:
- Edit the username.
- Submit the form and confirm that the changes are saved and displayed correctly.
3. Test the error-handling functionality by simulating invalid inputs or server errors.
### Background Context
The develop branch serves as the staging ground for features before they are merged into the production environment. This update includes key functionalities that enhance the user experience and profile management.

### Related Issues and PRs
Issue #21 : 
PR #20
